### ISSUE TITLE
Issues preprocessing the data in minfi

### ISSUE THREAD

**Issue by mininicovd** [chiaraherzog/IlluminaMouseMethylationmanifest#1]
Hello, 
I am currently trying to analyze IDAT files from the mouse methylation array but I am having trouble to preprocess the data.
The `preprocessFunnorm` and the `preprocessQuantile` functions from the minfi package don't work. 

> `preprocessFunnorm(RGset)` gives the error : `Error in dimnames(x) <- dn : length of 'dimnames' [2] not equal to array extent`

> `preprocessQuantile(RGset)` gives the error : `Error in normalize.quantiles(mat[Index2, ]) : Matrix expected in normalize.quantiles`

On the otherhand 

`preprocessNoob` seems to works, outputting a MethylSet object, while still giving this :  `Warning: useNames = NA is deprecated. Instead, specify either useNames = TRUE or useNames = TRUE.Warning: useNames = NA is deprecated. Instead, specify either useNames = TRUE or useNames = TRUE.`

Has anyone encountered this issue ? 
Has the minfi package combined with these mouse packages work for others ? 
I already analyzed my data using the sesame package which supported the mouse array but I now want to integrate my methylation data with expression data and I would like to use the MEAL package but it only supports methylation data preprocessed with minfi.
Hope someone will read this,
Thank you,
NVD

**Comment by chiaraherzog** [comment]
Hi @mininicovd,
apologies for the delayed reply! Thank you for pointing this out, it seems it has to do with the colData not being set (.e.g., sex). I will look into this now.
Of note, preprocessNoob does not seem to give an error for me on the latest minfi version.

I'll get back to you with an update soon.
Best, 
Chiara

**Comment by chiaraherzog** [comment]
Hi @mininicovd,
a quick update: there was a compatibility issue with Minfi. I have updated the minfi code to be compatible with the mouse manifest and generated a pull request.

While changes are being checked and will hopefully be available on Bioconductor soon, they are available here: https://github.com/chiaraherzog/minfi (install via devtools::install_github("chiaraherzog/minfi")

I tested this with the demo dataset provided from illumina and preprocessFunnorm and preprocessQuantiles now ran smoothly.

Note: the sex inference seems to not work as well for mice, but might be inherent to the algorithm and/or array (maybe different cutoffs can be tried), I have not investigated this in detail.
